src/streetrace/terminal_session.py

```diff
@@ -0,0 +1,785 @@
+"""Terminal session management with interactive command execution and automation.
```

Comment:
This huge documentation can become out of sync really quick. Do we really need such a big documentation here?